src/components/searchPage.js

```diff
@@ -1,24 +1,97 @@
+// SearchPage.js
+import React, { useState, useEffect } from "react";
 import LogoSection from "./logoSection";
 import ProductList from "./productList";
 import Header from "./header";
-import { formatProductListForSearchPage, getSummaryProductListAsync } from "../service/productService";
-import { useSelector } from "react-redux";
 import Footer from './footer';
-
-const productsToShow = await getSummaryProductListAsync();
+import ColorFilter from "./colorFilter";
+import {
+  formatProductListForSearchPage,
+  getSummaryProductListAsync,
+} from "../service/productService";
+import { useSelector } from "react-redux";
 
 function SearchPage() {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [productsToShow, setProductsToShow] = useState([]);
+  const [filteredProducts, setFilteredProducts] = useState([]);
```

Comment:
Let's rename `productsToShow` to `fullProductList` and rename `filteredProducts` to `productsToShow`.
This is because it could be confusing that the variable called `productsToShow` is not used as the list the user sees